agent/consul/stream/event_publisher.go

```diff
@@ -0,0 +1,305 @@
+package stream
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+// EventPublisher receives changes events from Publish, and sends them to all
+// registered subscribers.
+type EventPublisher struct {
+	// topicBufferSize controls how many trailing events we keep in memory for
+	// each topic to avoid needing to snapshot again for re-connecting clients
+	// that may have missed some events. It may be zero for no buffering (the most
+	// recent event is always kept though). TODO
+	topicBufferSize int
```

Comment:
This seems to be a relic that isn't used anywhere since I built the EventBuffer unbounded implementation?

Reply:
I think so ya, it is definitely not used, and from the description I suspected it wouldn't be. I will remove it. Thanks!